File: solutions/go/anagram/1/anagram.go
package anagram

import "strings"

func compareMaps(map1, map2 map[rune]int) bool {
    // Сравниваем длины карт
    if len(map1) != len(map2) {
        return false
    }

    // Сравниваем ключи и значения
    for key, value1 := range map1 {
        if value2, exists := map2[key]; !exists || value1 != value2 {
            return false
        }
    }

    return true
}

func Detect(subject string, candidates []string) []string {
	result := make([]string, 0)
    subject = strings.ToLower(subject)
    pattern := make(map[rune]int, 0)
    for _, v := range(subject) {
        pattern[v]++
    }
    for i := 0; i < len(candidates); i++ {
        word := strings.ToLower(candidates[i])
        if subject == word {
            continue
        }
        data := make(map[rune]int, 0)
        for _, v := range(word) {
        	data[v]++
    	}
        if compareMaps(pattern, data) {
            result = append(result, candidates[i])
        }
    }

    return result
}

File: solutions/python/anagram/1/anagram.py
def compare_dicts(dict1, dict2):
    if len(dict1) != len(dict2):
        return False

    for key, value1 in dict1.items():
        if dict2.get(key) != value1:
            return False

    return True

def find_anagrams(word, candidates):
    result = []
    subject = word.lower()
    from collections import Counter
    pattern = Counter(subject)
    for cand in candidates:
        word = cand.lower()
        if word == subject:
            continue
        data = Counter(word)
        if compare_dicts(pattern, data):
            result.append(cand)

    return result 